Sources/NetworkProtection/Diagnostics/NetworkProtectionConnectionTester.swift

```diff
@@ -169,23 +167,44 @@ final class NetworkProtectionConnectionTester {
 
     // MARK: - Timer scheduling
 
-    private func scheduleTimer() async {
+    private func scheduleTimer(testImmediately: Bool) async throws {
         await stopScheduledTimer()
 
+        await timerRunCoordinator.start()
+
+        if testImmediately {
+            do {
+                try await testConnection(isStartupTest: true)
+            } catch {
+                os_log("Rethrowing exception", log: log)
```

Comment:
I'd be inclined to remove this log, and let the caller who has to catch this error do the logging. Or we can just make it more clear like "Caught exception when testing connection", since this would be confusing to see on its own in the logs.